tests/testthat/test-trailing_blank_lines_linter.R

```diff
@@ -17,22 +17,38 @@ test_that("returns the correct linting", {
 
   # Construct a test file without terminal newline
   # cf. test-get_source_expressions.R
-  writeLines("lm(y ~ x)", tmp <- tempfile())
-  on.exit(unlink(tmp), add = TRUE)
-  writeBin(
-    # strip the last (two) element(s) (\r\n or \n)
-    head(
-      readBin(tmp, raw(), file.size(tmp)),
-      if (.Platform$OS.type == "windows") -2L else -1L
-    ),
-    tmp2 <- tempfile()
-  )
-  on.exit(unlink(tmp2), add = TRUE)
+  tmp <- withr::local_tempfile()
+  tmp2 <- withr::local_tempfile()
+  cat("lm(y ~ x)\n", file = tmp)
+  cat("lm(y ~ x)", file = tmp2)
 
   expect_lint(content = NULL, file = tmp, NULL, linter)
   expect_lint(content = NULL, file = tmp2, list(
     message = msg2,
     line_number = 1L,
     column_number = 10L
   ), linter)
+
+  # Construct an Rmd file without terminal newline
+  tmp3 <- withr::local_tempfile()
+  # TODO (@AshesITR): Remove \n escapes when #1406 is fixed
```

Comment:
nit: btw I think `TODO(#1406)` is better in general, then you can assign yourself the issue. makes it easier in case you decide to go on hiatus & not care about the issue for a few years :)